### ISSUE TITLE
Is it ok to reuse DbCompiledModel for different connection strings (the same server, identical db schema)

### ISSUE THREAD

**Issue by vladimir-cheverdyuk** [dotnet/efcore#22606]
I have question about old EF 6.0. We have a function
```
        public static T GetDataContext<T>(ConcurrentDictionary<string, DbCompiledModel> compiledModelCache, string connectionString, Func<string, DbConnection> getConnection, Action<DbModelBuilder> initDbModelBuilder)
        {
            var connection = getConnection(connectionString);
            DbCompiledModel compiledModel = null;

            if (!compiledModelCache.ContainsKey(connectionString))
            {
                var builder = new DbModelBuilder();
                initDbModelBuilder(builder);
                var model = builder.Build(connection);
                compiledModel = model.Compile();
                compiledModelCache.TryAdd(connectionString, compiledModel);
            }
            else
            {
                compiledModel = compiledModelCache[connectionString];
            }

            return ((T)Activator.CreateInstance(typeof(T), connection, compiledModel));
        }
```
that creates `DataContext` and `DbCompiledModel` based on connection string. We have a lot of tenants and all of them using identical db schema and they all on the same server. But we have a problem that our web site started consuming a lot of memory (2-3 Gb) when there are a lot of tenants. After Investigation of dump file I found that there are a lot of `DbCompiledModel` compiled queries etc because we have a static field `compiledModelCache` and all `DbCompiledModel` are there,
After checking code it looks like it is possible to reuse them. But I would like to confirm this with your guys.

Is it ok to change this code to this:
```
        public static T GetDataContext<T>(ConcurrentDictionary<string, DbCompiledModel> compiledModelCache, string connectionString, Func<string, DbConnection> getConnection, Action<DbModelBuilder> initDbModelBuilder)
        {
            var connection = getConnection(contextKey);
            DbCompiledModel compiledModel = null;

            var providerInvariantName = DbConfiguration.DependencyResolver.GetService<IProviderInvariantName>(DbProviderServices.GetProviderFactory(connection)).Name;
            var providerManifestToken = DbConfiguration.DependencyResolver.GetService<IManifestTokenResolver>().ResolveManifestToken(connection);
            string key = $"{providerInvariantName}|{providerManifestToken}";
            if (!compiledModelCache.ContainsKey(key))
            {
                var dbInfo = new DbProviderInfo(providerInvariantName, providerManifestToken);
                var builder = new DbModelBuilder();
                initDbModelBuilder(builder);
                var model = builder.Build(dbInfo);
                compiledModel = model.Compile();
                compiledModelCache.TryAdd(key, compiledModel);
            }
            else
            {
                compiledModel = compiledModelCache[key];
            }

            return ((T)Activator.CreateInstance(typeof(T), connection, compiledModel));
        }
```

Basically I'm creating model based on DbProviderInfo instead of connection string. I checked that inernaly it does exactly the same. After that it looks like DbCompiledModel is not depend on connection string and can be reused.
There no changes in data model or mapping based on connection string or anything else. Everything is always identical for every tenant.

I will appreciate any feedback.
